wg-naming/README.md

```diff
@@ -0,0 +1,70 @@
+<!---
+This is an autogenerated file!
+
+Please do not edit this file directly, but instead make changes to the
+sigs.yaml file in the project root.
+
+To understand how this file is generated, see https://git.k8s.io/community/generator/README.md
+--->
+# Naming Working Group
+
+
+## Stakeholder SIGs
+* SIG Architecture
+* SIG Contributor Experience
+* SIG Docs
+
+
+
+## Organizers
+
+* Celeste Horgan (**[@celestehorgan](https://github.com/celestehorgan)**), CNCF
+* Stephen Augustus (**[@justaugustus](https://github.com/justaugustus)**), VMware
+* Zach Corleissen (**[@zacharysarah](https://github.com/zacharysarah)**), Linux Foundation
+
+## Contact
+- Slack: [#wg-naming](https://kubernetes.slack.com/messages/wg-naming)
+- [Mailing list](https://groups.google.com/forum/#!forum/kubernetes-wg-naming)
+- [Open Community Issues/PRs](https://github.com/kubernetes/community/labels/wg%2Fnaming)
+<!-- BEGIN CUSTOM CONTENT -->
+
+**The following section will be reworked and formalized as a charter once the
+Working Group has been approved by the Steering Committee.**
+
+### Goals
+
+- Evaluate language and naming choices within the Kubernetes organization, with
+  a specific initial focus of:
+  - Removing language which cannot be fully separated from racist, sexist or
+    homophobic connotations in cultural contexts outside of computer
+    programming
+  - Improving clarity of codebases and documentation by replacing idioms,
+    metaphors and slang specific to the English language
+- Create a list of terms to avoid and propose alternatives
+- Define a process by which any member of the Kubernetes organization can
+  propose a language recommendation, have them evaluated based on a rubric, and
+  see them implemented across all codebases.
+  - Provide an easily findable location for language recommendations and
+    follow-up issues, similar to an architectural decision record.
+- Work with stakeholder SIGs to implement the changes recommended. We
+  anticipate the following:
+  - Provide stakeholder SIGs with guidance on naming, language conventions, and
+    processes
+  - Collaborate with SIG Architecture and other stakeholders on an
+    implementation timeline and strategies for dealing with follow-up issues
+    from renaming, like deprecations
+  - Collaborate with SIG Docs and SIG Contributor Experience on documenting
+    language recommendations and processes
+  - Work with the Code of Conduct Committee to add code architecture to the COC
+
+### Dissolution Criteria
+
+When this WG dissolves, the Kubernetes community should have:
+
+- A process in place to evaluate language changes on an ongoing basis
+- A list of terms to avoid in codebases across the organization
+- A timeline on which to replace component names in the kubernetes/kubernetes
+  codebase
```

Comment:
Few questions here:

- Would this WG focus only on k/k? My understanding was that it would look at naming practices across all our repos (even k-sigs, k-csi, k-client), docs, etc.

- Maybe it would be better to rename "component names" to something slightly more...generic? Just to ensure that we don't tie the dissolution criteria to only component names... :thinking: 

- Would the WG disband after a _timeline_ has been agreed upon or would it disband after all the changes have been made as per the timeline?

Reply:
FWIW, I personally would find it helpful if the group stuck around until changes have been made.  The model I have is WG-Apply, which existed until server side apply was implement.

Reply:
I haven't conferred with @justaugustus on this, so he may disagree, but here's my thoughts:

> Would this WG focus only on k/k? My understanding was that it would look at naming practices across all our repos (even k-sigs, k-csi, k-client), docs, etc.

Per an earlier comment, we're changing the wording in this WG to address the Kubernetes _project_ – so presumably.. everything? 😅 

> Maybe it would be better to rename "component names" to something slightly more...generic? Just to ensure that we don't tie the dissolution criteria to only component names... 🤔

+1! "feature names?" "method names"? I'm unsure. I'll brainstorm on this for a bit.

> Would the WG disband after a timeline has been agreed upon or would it disband after all the changes have been made as per the timeline?

Personally I'd like to stick around until work is executed for accountability's sake. This isn't high priority work in relation to new releases, for example, but I think it's high importance and I think having the WG around to see the work through would be good.

Reply:
(Added as a post-formation discussion point.)

Reply:
> I haven't conferred with @justaugustus on this, so he may disagree, but here's my thoughts:
> 
> > Would this WG focus only on k/k? My understanding was that it would look at naming practices across all our repos (even k-sigs, k-csi, k-client), docs, etc.
> 
> Per an earlier comment, we're changing the wording in this WG to address the Kubernetes _project_ – so presumably.. everything? sweat_smile
> 
> > Maybe it would be better to rename "component names" to something slightly more...generic? Just to ensure that we don't tie the dissolution criteria to only component names... thinking
> 
> +1! "feature names?" "method names"? I'm unsure. I'll brainstorm on this for a bit.
> 
> > Would the WG disband after a timeline has been agreed upon or would it disband after all the changes have been made as per the timeline?
> 
> Personally I'd like to stick around until work is executed for accountability's sake. This isn't high priority work in relation to new releases, for example, but I think it's high importance and I think having the WG around to see the work through would be good.

Agreed that the goal is to serve across the project, not just kubernetes/kubernetes.



> > Maybe it would be better to rename "component names" to something slightly more...generic? Just to ensure that we don't tie the dissolution criteria to only component names... thinking
> 
> +1! "feature names?" "method names"? I'm unsure. I'll brainstorm on this for a bit.

Yep. We can workshop the language post-formation.



> Personally I'd like to stick around until work is executed for accountability's sake. This isn't high priority work in relation to new releases, for example, but I think it's high importance and I think having the WG around to see the work through would be good.

Agreed. I also intend to action on changes ahead of WG dissolution.